src/context.rs

```diff
@@ -565,11 +560,18 @@ impl DefaultSeafowlContext {
     // Check that the TableReference doesn't have a database/schema in it.
     // We create all external tables in the staging schema (backed by DataFusion's
     // in-memory schema provider) instead.
-    fn resolve_staging_ref<'a>(
-        &'a self,
-        name: &'a str,
-    ) -> Result<ResolvedTableReference<'a>> {
-        let reference: TableReference = name.into();
+    fn resolve_staging_ref(
+        &self,
+        name: &OwnedTableReference,
+    ) -> Result<OwnedTableReference> {
+        // NB: Since Datafusion 16.0.0 there's this OwnedTableReference enum and for external tables
+        // the parsed ObjectName (which may be multipart, fully-qualified name) is coerced into the
+        // `Bare` enum variant (see `external_table_to_plan` in datafusion-sql) for some reason.
+        //
+        // This means that any potential catalog/schema references get condensed into the name, so
+        // we have to unravel that name here again, and then resolve it properly.
+        let full_name = name.to_string();
+        let reference = TableReference::from(full_name.as_str());
```

Comment:
Not sure if this as an issue: If I have a table called `my.table`, then could this incorrectly be parsed as `my` being the collection / schema? Disregard if "." is not a valid char even within quotes for a table name.

Reply:
Good point. I just tried it out and indeed we there is this problem. Namely if I do
```sql
CREATE EXTERNAL TABLE \"my.table\"
...
```
The parsed reference has `default` for the catalog, but `"my` as the schema and `table"` as the table name.